### ISSUE TITLE
java.lang.NoClassDefFoundError: fish/payara/notification/eventbus/EventbusMessage

### ISSUE THREAD

**Issue by mkarg** [payara/Payara#1633]
# Description #
----------

javax.ejb.embeddable.EJBContainer.createEJBContainer() (also GlassFishRuntime.bootstrap().newGlassFish().start();
) throws exception that fish.payara.notification.eventbus.EventbusMessage is not found.

## Expected Outcome

Payara Embedded starts a container instance

## Current Outcome

```
Mai 28, 2017 6:07:02 PM com.sun.enterprise.v3.server.AppServerStartup$MasterRunLevelListener onError
INFORMATION: Shutdown requested
java.lang.NoClassDefFoundError: fish/payara/notification/eventbus/EventbusMessage
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:114)
        at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:125)
        at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
        at sun.reflect.generics.visitor.Reifier.reifyTypeArguments(Reifier.java:68)
        at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:138)
        at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
        at sun.reflect.generics.repository.ClassRepository.getSuperclass(ClassRepository.java:90)
        at java.lang.Class.getGenericSuperclass(Class.java:777)
        at org.glassfish.hk2.utilities.reflection.ReflectionHelper.typesFromSubClassToDeclaringClass(ReflectionHelper.java:205)
        at org.glassfish.hk2.utilities.reflection.ReflectionHelper.resolveMember(ReflectionHelper.java:166)
        at org.glassfish.hk2.utilities.reflection.ReflectionHelper.resolveField(ReflectionHelper.java:153)
        at org.jvnet.hk2.internal.Utilities.getFieldInjectees(Utilities.java:1861)
        at org.jvnet.hk2.internal.ClazzCreator.initialize(ClazzCreator.java:160)
        at org.jvnet.hk2.internal.ClazzCreator.initialize(ClazzCreator.java:179)
        at org.jvnet.hk2.internal.SystemDescriptor.internalReify(SystemDescriptor.java:723)
        at org.jvnet.hk2.internal.SystemDescriptor.reify(SystemDescriptor.java:678)
        at org.jvnet.hk2.internal.ServiceLocatorImpl.reifyDescriptor(ServiceLocatorImpl.java:459)
        at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:1986)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:114)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:88)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.oneJob(CurrentTaskFuture.java:1213)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.run(CurrentTaskFuture.java:1144)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpOneLevel.run(CurrentTaskFuture.java:762)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpAllTheWay.go(CurrentTaskFuture.java:561)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpAllTheWay.access$100(CurrentTaskFuture.java:443)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:154)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:149)
        at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.proceedTo(AsyncRunLevelContext.java:436)
        at org.glassfish.hk2.runlevel.internal.RunLevelControllerImpl.proceedTo(RunLevelControllerImpl.java:71)
        at com.sun.enterprise.v3.server.AppServerStartup.proceedTo(AppServerStartup.java:534)
        at com.sun.enterprise.v3.server.AppServerStartup.run(AppServerStartup.java:331)
        at com.sun.enterprise.v3.server.AppServerStartup.doStart(AppServerStartup.java:228)
        at com.sun.enterprise.v3.server.AppServerStartup.start(AppServerStartup.java:219)
        at com.sun.enterprise.glassfish.bootstrap.GlassFishImpl.start(GlassFishImpl.java:81)
        at org.glassfish.ejb.embedded.EJBContainerImpl.<init>(EJBContainerImpl.java:100)
        at org.glassfish.ejb.embedded.EJBContainerProviderImpl.createContainer(EJBContainerProviderImpl.java:221)
        at org.glassfish.ejb.embedded.EJBContainerProviderImpl.createEJBContainer(EJBContainerProviderImpl.java:129)
        at javax.ejb.embeddable.EJBContainer.createEJBContainer(EJBContainer.java:127)
        at javax.ejb.embeddable.EJBContainer.createEJBContainer(EJBContainer.java:102)
        at SmokeIT.testshouldScanForJaxRsApplications(SmokeIT.java:8)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: java.lang.ClassNotFoundException: fish.payara.notification.eventbus.EventbusMessage
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 78 more
```

## Steps to reproduce (Only for bug reports) 

Either perform `EJBContainer.createEJBContainer();` or `GlassFishRuntime.bootstrap().newGlassFish().start();`.
## Environment ##

- **Payara Version**: 4.1.2.172
- **Edition**: Embedded-All
- **JDK Version**: OracleJRE 7u121
- **Operating System**: Windows
- **Database**: none

**Comment by fturizo** [comment]
@mkarg, starting from release _4.1.1.171_, some of the classes used for Payara specific features like the request tracing service and the CDI event bus have been separated into a new [Public API](https://docs.payara.fish/v/next/documentation/extended-documentation/app-deployment/public-api.html) that simplifies dependencies on the development of applications. 

You need to the following Maven dependency to your POM file to correctly use the `payara-embedded-all `dependency:

```xml
<dependency>
    <groupId>fish.payara.api</groupId>
    <artifactId>payara-api</artifactId>
    <version>4.1.1.171</version>
    <scope>provided</scope>
</dependency>
```

**Comment by mkarg** [comment]
@fturizo I do **not** want to use any additional _Payara specific_ features. I simply want to start an embedded EJB container, which is a Java EJB _core_ feature. So does that mean, to use payara-embedded-all, one _always_ has to add _two_ dependencies to his POM? That is pretty annoying and counter-intuitive!

**Comment by mkarg** [comment]
I confirm that by adding the above dependency it works well.

**Comment by smillidge** [comment]
Looks like we need to ensure api is packaged into the embedded jars.

**Comment by fturizo** [comment]
You are correct @mkarg. As @smillidge suggested, I've opened the internal issue `PAYARA-1733` so this change is implemented to prevent this confusion.

**Comment by smillidge** [comment]
Fixed by #1660
